autosklearn/evaluation/abstract_evaluator.py

```diff
@@ -361,23 +370,22 @@ def file_output(
                     },
                 )
 
+        # Abort if we don't want to output anything
+        # note: since disable_file_output can be a list, we have to explicitly
+        # compare it against True
+        if self.disable_file_output == True:
+            return None, {}
```

Comment:
The original code checked always ` if self.disable_file_output != True` before writing each file. I changed this a bit so that we check just once if we need to write anything at all. In this way, we get simpler `if` statements for each file.
 Let me know if you see any problems with that.